src/main/java/org/folio/service/processing/reader/EdifactReader.java

```diff
@@ -0,0 +1,77 @@
+package org.folio.service.processing.reader;
+
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.xlate.edi.stream.EDIInputFactory;
+import io.xlate.edi.stream.EDIStreamEvent;
+import io.xlate.edi.stream.EDIStreamException;
+import io.xlate.edi.stream.EDIStreamReader;
+import org.folio.rest.jaxrs.model.InitialRecord;
+import org.folio.rest.jaxrs.model.RecordsMetadata;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation reads source records in EDIFACT format from the local file system.
+ * <code>next</code> method returns list of {@link InitialRecord} when the target file has come to the end.
+ */
+public class EdifactReader implements SourceReader {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(EdifactReader.class);
+  private static final Charset DEFAULT_CHARSET = StandardCharsets.ISO_8859_1;
+
+  private EDIStreamReader reader;
+  private EdifactParser edifactParser;
+  private File file;
+  private Map<String, Character> delimiters = null;
+
+  public EdifactReader(File file) {
+    try {
+      this.file = file;
+      EDIInputFactory factory = EDIInputFactory.newFactory();
+      InputStream stream = new FileInputStream(file);
+      reader = factory.createEDIStreamReader(stream);
+      if (reader.next() == EDIStreamEvent.START_INTERCHANGE) {
+        delimiters = reader.getDelimiters();
+        edifactParser = new EdifactParser(delimiters);
+      }
+    } catch (IOException | EDIStreamException e) {
+      String errorMessage = "Can not initialize reader. Cause: " + e.getMessage();
+      LOGGER.error(errorMessage);
+      throw new IllegalArgumentException(errorMessage);
```

Comment:
we have a custom Reader exception that is used as a wrapper over the IOException, I think it is applicable here as well

Reply:
Fixed.